### PR TITLE
V3.6.0 add instanced particle

### PR DESCRIPTION
**Pull request by zxx43**
Re: #

Changelog:
 *

### CODE REVIEW

File: cocos/particle/renderer/particle-system-renderer-base.ts [hunk 1]
```diff
@@ -60,16 +60,28 @@ export interface IParticleSystemRenderer {
...
+    setUseInstance (value: boolean): void;
+    getUseInstance (): boolean;
 }
 
 export abstract class ParticleSystemRendererBase implements IParticleSystemRenderer {
     protected _particleSystem: any = null;
     protected _model: ParticleBatchModel | null = null;
     protected _renderInfo: ParticleSystemRenderer | null = null;
     protected _vertAttrs: Attribute[] = [];
+    protected _useInstance;
```
Comment (star-e) [inline cocos/particle/renderer/particle-system-renderer-base.ts]: lint problem

Comment (star-e) [inline cocos/particle/renderer/particle-system-renderer-base.ts:72]: lint problem

**Comment by star-e** [comment]
Currently we have used _useInstance to differ plain/instanced rendering strategies. It is ok for current situation.
However, if we need to add new features in the future, ParticleBatchModel will need some refactoring.

**Comment by zxx43** [comment]
> Currently we have used _useInstance to differ plain/instanced rendering strategies. It is ok for current situation. However, if we need to add new features in the future, ParticleBatchModel will need some refactoring.

Split current ParticleBatchModel into BatchModel & InstanceModel

**Review by star-e — APPROVED** [review]

**Pull request merged**